### ISSUE TITLE
Theory: Back-propagation and Neural ODEs

### ISSUE THREAD

**Issue by jaamestaay** [jaamestaay/M2R-Group-29#5]

**Comment by LaiquendiJ** [comment]
Julia implementation for neural network and neural ODE https://scholarworks.arcadia.edu/cgi/viewcontent.cgi?article=1133&context=showcase
Python implementation for neural ODE https://github.com/msurtsukov/neural-ode

**Comment by LaiquendiJ** [comment]
<img width="369" alt="Screen Shot 2024-05-30 at 18 13 25" src="https://github.com/jaamestaay/M2R-Group-29/assets/105692121/779aa0b9-c0e2-4c96-bc12-e670afe22235">

I don't understand why the forward process is smooth, but for backpropagation, we need to update this particular gradient at each timestep.(the screenshot is from https://arxiv.org/pdf/1806.07366)
